[AIRFLOW-6181] Add InProcessExecutor (#6740)
mistercrunch/Airflow: docs/executor/inprocess.rst

 .. Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

InProcess Executor
==================

The ``InProcessExecutor`` is meant as a debug tool and can be used from IDE.
It is a single process executor that queues :class:`~models.taskinstance.TaskInstance`
and executes them by running ``_run_raw_task`` method.

> Due to it's nature the executor can be sued with SQL Lite database. When used

Reviewer: ```suggestion
Due to its nature the executor can be used with all databases - including SQLite database. When used
```

> with sensors the executor will change sensor mode to ``reschedule`` to do not

Reviewer: ```suggestion
with sensors the executor will change sensor mode to ``reschedule`` to avoid blocking
```

> block the execution of DAG.

Reviewer: ```suggestion
the execution of DAG.
```

Additionally ``InProcessExecutor`` can be used in a fail-fast mode that will make

> DAG to fail after first failed task. To enable this option set ``AIRFLOW__DEBUG__FAIL_FAST=True``

Reviewer: ```suggestion
all other running or scheduled tasks fail immediately after first failed task. 
To enable this option set ``AIRFLOW__DEBUG__FAIL_FAST=True``
```

> or adjust this setting in your ``airflow.cfg``.

Reviewer 2: ```suggestion
or adjust option ``fail_fast`` in your ``airflow.cfg``.
```

**IDE setup steps:**

> 1. Add main at the end of your DAG file to make it runnable:

Reviewer: ```suggestion
1. Add `main` method at the end of your DAG file to make it runnable:
```

  .. code-block:: python

    if __name__ == '__main__':
      dag.clear(reset_dag_runs=True)
      dag.run()

>    By adding this running a DAG file will run a backfill job.

Reviewer: ```suggestion
   When you add those lines, running a DAG file will run a backfill job.
```

2. Setup ``AIRFLOW__CORE__EXECUTOR=InProcessExecutor`` in run configuration of your IDE. In
   this step you should also setup all environment variables required by your DAG.

3. Run the DAG file.

> Additionally ``InProcessExecutor`` can be used in a fail-fast mode that will make

Reviewer: It's repeated from the previous chapter. I think it can be removed from here..
